web/src/search/stream.tsx

```diff
@@ -162,6 +208,14 @@ export function search(
                 observer.next({ type: 'filematches', matches: JSON.parse(event.data) as FileMatch[] })
             })
         )
+        subscriptions.add(
+            fromEvent(eventSource, 'repomatches').subscribe((event: Event) => {
+                if (!(event instanceof MessageEvent)) {
+                    throw new TypeError('internal error: expected MessageEvent in streaming search filematches')
```

Comment:
```suggestion
                    throw new TypeError('internal error: expected MessageEvent in streaming search repomatches')
```

Alternatively, since we're doing the `if (!(event instanceof MessageEvent)) { }` dance three times already, here is a helper suggestion that would help DRY:

```typescript
const observeMessages = <T extends {}>(eventSource: EventSource, eventName: SearchEvent['type']): Observable<T> =>
    fromEvent(eventSource, eventName).pipe(
        map((event: Event) => {
            if (!(event instanceof MessageEvent)) {
                throw new TypeError('internal error: expected MessageEvent in streaming search ${eventName}')
            }
            try {
                const parsedData = JSON.parse(event.data) as T
                return parsedData
            } catch (err) {
                throw new Error(`Could not parse ${eventName} message data in streaming search`)
            }
        })
    )

subscriptions.add(
    observeMessages<RepositoryMatch[]>(eventSource, 'repomatches').pipe(
        map(matches => ({ type: 'repomatches' as const, matches }))
    ).subscribe(observer)
)
```

Reply:
`SearchEvent['type']` wow, nifty stuff. Is there a way to annotate the could not parse error with information about the original parsing error?

For bonus marks I am sure there is a way we can make observeMessages include `type` field in the response in a typesafe way.